parquet-column/src/main/java/org/apache/parquet/column/values/delta/DeltaBinaryPackingValuesWriter.java

```diff
@@ -130,6 +135,21 @@ public long getBufferedSize() {
 
   @Override
   public void writeInteger(int v) {
+    /*
+     * isInt is needed. Just calling internalWrite might result in a different binary representation.
+     * Example: [Integer.MIN_VALUE, Integer.MAX_VALUE, Integer.MIN_VALUE, Integer.MAX_VALUE]
+     */
+    isInt = true;
```

Comment:
Calling both writeInteger and writeLong would corrupt data. I think that the object model calls to MessageColumnIORecordConsumer determine whether writeInteger or writeLong is used, and that makes it possible for an object model to mix calls.

What about making this writer abstract and implementing each method separately? The code would be nearly identical, but would throw an exception if it is improperly used.